lib/formats/odata.js

```diff
@@ -418,6 +430,48 @@ const singleRowToOData = (fields, row, domain, originalUrl, query) => {
   });
 };
 
+const systemFields = new Set([
+  '__system/submissiondate',
+  '__system/updatedat',
+  '__system/submitterid',
+  '__system/submittername',
+  '__system/attachmentspresent',
+  '__system/attachmentsexpected',
+  '__system/status',
+  '__system/reviewstate',
+  '__system/deviceid',
+  '__system/edits',
+  '__system/formversion',
+]);
+
+// Validates $select query parameter including metadata properties and returns list of FormFields
+const filterFields = (formFields, select, table) => {
+  const filteredFields = new Set();
+  const fieldMap = formFields.reduce((map, field) => ({ ...map, [field.path.toLowerCase()]: field }), {});
+  for (const property of select.split(',').map(p => p.trim().toLowerCase())) {
+    // validate metadata properties. __system/.. properties are only valid for Submission table
+    if (property.startsWith('__')) {
+      if (!(property === '__id' || (table === 'Submissions' && systemFields.has(property))))
+        throw Problem.user.propertyNotFound({ field: property });
+    } else {
+      // we have to include parents fields in the result to handle grouped fields
+      const propSegments = property.split('/');
+      let path = '';
+      for (let i = 0; i < propSegments.length; i+=1) {
+        path += `/${propSegments[i]}`;
+        const field = fieldMap[path];
+        if (!field) throw Problem.user.propertyNotFound({ field: property });
+        // it's ok to include field with repeat type so that user can have navigation link
+        // but child field of a repeat field is not supported
+        if (field.type === 'repeat' && i < propSegments.length - 1) throw Problem.user.unsupportedODataSelectField({ field: property });
+        filteredFields.add(field);
```

Comment:
I'm not totally sure, but I think that if the user `$select`s a field in a group, but doesn't `$select` the group, we should also add the group to `filteredFields`. For example: `$select=meta/instanceID` instead of `$select=meta,meta/instanceID`. Otherwise I think `generateDateFrame()` won't work as expected.

Reply:
This is handled by adding group fields automatically to `filteredFields`

Reply:
Tested in `should return a single row result with selected properties` integration test